src/dlls/mscoree/mscoree.cpp

```diff
@@ -1188,15 +1165,12 @@ STDAPI LoadLibraryShimInternal(LPCWSTR szDllName, LPCWSTR szVersion, LPVOID pvRe
         }
         return S_OK;
     }
-
-#endif // FEATURE_CORECLR
-
 }
-
-#endif // CROSSGEN_COMPILE
+#endif
+#endif 
 
 static DWORD g_dwSystemDirectory = 0;
-static WCHAR g_pSystemDirectory[MAX_LONGPATH + 1];
+static WCHAR * g_pSystemDirectory = new WCHAR[MAX_LONGPATH+1];
```

Comment:
On initial assigment in ```SetInternalSystemDirectory``` if this pointer doesn't point to allocated memory wouldn't memcpy fail?